stream_alert/rule_processor/handler.py

```diff
@@ -86,6 +88,12 @@ def s3_process(self, payload, classifier):
             classifier.classify_record(payload, data)
             self.process_alerts(payload)
 
+    def sns_process(self, payload, classifier):
+        """Process SNS data for alerts"""
+        data = StreamPreParsers.pre_parse_sns(payload.raw_record)
```

Comment:
I made a small change, but it just mostly consolidates duplicate code into a method. The `pre_parse..` methods pull data from different keys, so this is about as simple as this get (see `pre_parsers.py` for more context).